mypyc/genops.py

```diff
@@ -494,46 +478,93 @@ def f(self, x: object) -> int: ...
                       cls.name, self.module_name,
                       FuncSignature(rt_args, ret_type), blocks, env)
 
+    def contains_func_def(self, stmt: Statement) -> bool:
+        """Traverses a Statement recursively using BFS and checks if it or any nested statements
+        contain a FuncDef.
+        """
+        if isinstance(stmt, (FuncItem, ForStmt, WhileStmt)):
+            if (any(isinstance(s, FuncDef) for s in stmt.body.body) or
+                    any(self.contains_func_def(s) for s in stmt.body.body)):
```

Comment:
Maybe merge the checks into one `any`? Could be cleaner.

Reply:
I left them as two `any` statements because I want to traverse an entire level before diving deeper into the next level.